contracts/staking/EpochSelector.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./interfaces/IEpochSelector.sol";
+import "./ClusterSelector_wih_abi_encode.sol";
+
+/// @title Contract to select the top 5 clusters in an epoch
+contract EpochSelector is AccessControl, ClusterSelector, IEpochSelector {
+    /// @notice Event emitted when Cluster is selected
+    /// @param epoch Number of Epoch
+    /// @param cluster Address of cluster
+    event ClusterSelected(uint256 indexed epoch, address indexed cluster);
+
+    /// @notice length of epoch
+    uint256 public constant epochLength = 4 hours;
+
+    /// @notice timestamp when the selector starts
+    uint256 public immutable startTime;
+
+    /// @notice Number of clusters selected in every epoch
+    uint256 public immutable numberOfClustersToSelect;
+
+    /// @notice clusters selected during each epoch
+    mapping(uint256 => address[]) public clustersSelected;
+
+    /// @notice ID for update role
+    bytes32 public constant updaterRole = keccak256(abi.encode("updater")); // find standard format for this
+
+    /// @notice ID for updater admin role
+    bytes32 public constant updaterAdminRole = keccak256(abi.encode("updater admin")); // find standard format for this
+
+    constructor(address _admin, uint256 _numberOfClustersToSelect) ClusterSelector() {
```

Comment:
let's take startTime also as an input so that we have more control on when epochs are going to start